packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/AccessorModifierIrGeneration.kt

```diff
@@ -1017,14 +1017,27 @@ class AccessorModifierIrGeneration(private val pluginContext: IrPluginContext) {
             val isNullable = collectionGenericType.isNullable()
 
             // Lists of objects/embedded objects and sets of object may NOT contain null values, but dictionaries may
-            if (collectionType == CollectionType.SET || collectionType == CollectionType.LIST) {
-                if (isNullable) {
-                    logError(
-                        "Error in field ${declaration.name} - ${collectionType.description} does not support nullable realm objects element types.",
-                        declaration.locationOf()
-                    )
-                    return null
+            when (collectionType) {
```

Comment:
I had to make changes to this to fix an error that allowed `RealmDictionary<RealmObject>` fields in compile time which is incorrect since dictionaries of objects/embedded objects can only be nullable. Check also changes related to `ListTests`, `SetTests` and `DictionaryTests` as well.